File: sdk/alcedo-sdk/src/items.ts
export function createItemsResource(ky: any) {
  return {
    list: (name: string, params?: Record<string, string>) =>
      ky.get(`items/${encodeURIComponent(name)}`, { searchParams: params }).json(),

    create: (name: string, data: any) =>
      ky.post(`items/${encodeURIComponent(name)}`, { json: data }).json(),

    update: (name: string, data: { filter: any; update: any }) =>
      ky.put(`items/${encodeURIComponent(name)}`, { json: data }).json(),

    delete: (name: string, data: { filter?: any; pk_values?: any[] }) =>
      ky.delete(`items/${encodeURIComponent(name)}`, { json: data }).json(),

    get: (name: string, id: string, params?: Record<string, string>) =>
      ky.get(`items/${encodeURIComponent(name)}/${encodeURIComponent(id)}`, { searchParams: params }).json(),

    patch: (name: string, id: string, data: any) =>
      ky.patch(`items/${encodeURIComponent(name)}/${encodeURIComponent(id)}`, { json: data }).json(),

    query: (name: string, data: any) => {
      const params = new URLSearchParams()
      if (data.filter) params.set('filter', JSON.stringify(data.filter))
      if (data.limit) params.set('limit', String(data.limit))
      if (data.offset) params.set('offset', String(data.offset))
      if (data.fields) params.set('fields', data.fields.join(','))
      if (data.sort && data.sort.length > 0) {
        params.set('sort', data.sort[0].field)
        if (data.sort[0].order || data.sort[0].direction) {
          params.set('order', data.sort[0].order || data.sort[0].direction)
        }
      }
      return ky.get(`items/${encodeURIComponent(name)}`, { searchParams: params }).json()
    },

    grouped: (name: string, data: any) =>
      ky.post(`items/${encodeURIComponent(name)}/grouped`, { json: data }).json(),

    references: (name: string, id: string) =>
      ky.get(`items/${encodeURIComponent(name)}/${encodeURIComponent(id)}/references`).json(),
  };
}

File: sdk/alcedocore-sdk-python/alcedo_sdk/resources/items.py
import json
from typing import Any, Optional

from .base import BaseResource


class ItemsResource(BaseResource):
    """Collection items CRUD operations. Maps to /api/items/{name} endpoints."""

    async def list(
        self,
        name: str,
        params: Optional[dict[str, str]] = None,
    ) -> dict:
        return await self._request("GET", f"/api/items/{name}", params=params)

    async def create(
        self,
        name: str,
        data: Any,
    ) -> dict:
        return await self._request("POST", f"/api/items/{name}", json=data)

    async def update(
        self,
        name: str,
        data: dict,
    ) -> dict:
        return await self._request("PUT", f"/api/items/{name}", json=data)

    async def delete(
        self,
        name: str,
        data: dict,
    ) -> dict:
        return await self._request("DELETE", f"/api/items/{name}", json=data)

    async def get(
        self,
        name: str,
        id: str,
        params: Optional[dict[str, str]] = None,
    ) -> dict:
        return await self._request("GET", f"/api/items/{name}/{id}", params=params)

    async def patch(
        self,
        name: str,
        id: str,
        data: dict,
    ) -> dict:
        return await self._request("PATCH", f"/api/items/{name}/{id}", json=data)

    async def query(
        self,
        name: str,
        data: dict,
    ) -> dict:
        params = {}
        if "filter" in data:
            params["filter"] = json.dumps(data["filter"])
        if "limit" in data:
            params["limit"] = str(data["limit"])
        if "offset" in data:
            params["offset"] = str(data["offset"])
        if "fields" in data:
            params["fields"] = ",".join(data["fields"])
        if "sort" in data and data["sort"]:
            params["sort"] = data["sort"][0]["field"]
            params["order"] = data["sort"][0].get("order") or data["sort"][0].get("direction", "asc")
        return await self._request("GET", f"/api/items/{name}", params=params)

    async def grouped(
        self,
        name: str,
        data: dict,
    ) -> dict:
        return await self._request("POST", f"/api/items/{name}/grouped", json=data)

    async def references(
        self,
        name: str,
        id: str,
    ) -> dict:
        return await self._request("GET", f"/api/items/{name}/{id}/references")
